contracts/PoolManager.sol

```diff
@@ -479,8 +481,9 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
     {
         if (msg.sender != owner && msg.sender != address(protocolFeeController)) revert InvalidCaller();
 
-        amountCollected = (amount == 0) ? protocolFeesAccrued[currency] : amount;
-        protocolFeesAccrued[currency] -= amountCollected;
+        amountCollected = (amount == 0) ? balanceOf(address(protocolFeeController), currency.toId()) : amount;
+
+        _burn(address(protocolFeeController), currency.toId(), amountCollected);
```

Comment:
I'm not an official reviewer so I could be wrong. I just want someone more knowledgeable to take a look.

I wonder if we want to call `_burn` when `msg.sender == address(protocolFeeController)` since calling `currency.transfer` below.

Reply:
Thanks for the review!

In this singleton design, all ERC20s are held by the Pool Manager contract. Individual pool balances of ERC20s (aka currencies) are being accounted for within this Pool Manager contract.  

This internal call `burn()` is decreasing the ERC1155 balance (or "internal" balance) of the protocol fee controller's fee token (some ERC20). Then, `currency.transfer` is transferring (`ERC20.transfer()`) the fee token rom the pool manager to the (external) protocol fee controller.

Let me know if that makes sense or if I'm misunderstanding something! Thanks!